packages/tooling/fast-tooling/src/web-components/html-render-layer-inline-edit/inline-edit.spec.md

```diff
@@ -0,0 +1,33 @@
+# Abstract
+This spec outlines the functionality of the inline editing layer for use with the `HTMLRender` web component.
+
+## Overview
+This component will derive from the `HTMLRenderLayer` class and when placed in a `HTMLRender` component will allow
+for inline editing of text. 
+
+## Interaction
+Inline text editing will be initiated when the user double-clicks on a text node. When this happens the `HTMLRenderLayerInlineEdit`
+control will create a `textarea` and attempt to match the size, position and style so that the `textarea` input is "invisible" against
+the background element with the exception of a thin border so that to the user it appears that they are typing directly into the container 
+element. While the user is typing within this area keyboard/mouse interactions will work normally (letters, numbers, special characters, 
+arrow key navigation, mouse click curser placement, highlighting, copy/paste etc.) with the exceptions of Tab, Enter and Esc. Tab and Enter 
+will perform a confirmation action which will remove the `textarea`, replace the text node with the new text and transmit a data message
+announcing the update. The Esc key will cancel the text edit, removing the `textarea` and restoring the text node to the original value.
+
+## Implementation
+When the `HTMLRender` component detects a `dblclick` event it determine if the even occured on a text node and walk up the `composedPath` 
+until it finds the containing element. It will then send a `ActivityType.DoubleClick` alert to all of the child layers. 
+The `HTMLRenderLayerInlineEdit` layer will respond to this alert by finding the position and styles (background color, font color, size, family etc)
+and will create a `textarea` input element with the same styles placed directly over the target location. It will also save a backup of the 
```

Comment:
> directly over the target location

Hmmm, is there a concern that adding or removing lines could move the content below up/down the page (or should)? If we don't do this do we think it will be a problem, especially in the case of removing lines. I suppose if the `textarea` does not resize smaller, we won't have an issue.

Reply:
I mention in the Implementation section that the text node will be updated as the user types and the textarea will resize to match. So it will resize as lines are added/removed but I should probably mention that it would need some minimum size. Especially in situations like when an element with no text is double clicked.